handler/src/main/java/io/netty/handler/ssl/SslContextBuilder.java

```diff
@@ -48,6 +51,17 @@ public static SslContextBuilder forServer(File keyCertChainFile, File keyFile) {
     /**
      * Creates a builder for new server-side {@link SslContext}.
      *
+     * @param key a PKCS#8 private key
+     * @param keyCertChain the X.509 certificate chain
+     * @see #keyManager(PrivateKey, X509Certificate[])
+     */
+    public static SslContextBuilder forServer(PrivateKey key, X509Certificate... keyCertChain) {
+        return new SslContextBuilder(true).keyManager(key, keyCertChain);
```

Comment:
I understand where you are coming from. The point I am trying to get at is that it is a slippery slope. Having clear responsibilities of who is responsible for bounds checking (or null checking) can prevent everyone having to check for it.  If bounds change, and multiple places are doing the bounds checks, then you have to figure out everywhere that duplicated the now incorrect bounds.

The exception / stack trace may be a few buried in a few more frames, but the exception can provide a string message which provides context.